core/src/main/java/amino/run/policy/scalability/ScaleUpFrontendPolicy.java

```diff
@@ -36,15 +35,20 @@ public void setReplicationRateInMs(int replicationRateInMs) {
 
         @Override
         public boolean equals(Object o) {
-            if (this == o) return true;
-            if (o == null || getClass() != o.getClass()) return false;
+            if (this == o) {
```

Comment:
Not your fault, but neither equals() nor hashCode are correct:

https://stackoverflow.com/questions/2066917/overriding-equals-hashcode-in-sub-classes-considering-super-fields

Reply:
ok, I will check this and modify

Reply:
`ScaleUpFrontendPolicy.Config` do not inherit `LoadBalancedFrontendPolicy.Config`. They are independent classes to hold their respective config. So, this generation of `equals()` and `hashCode()` looks correct to me.